src/scripts/items/ItemHandler.ts

```diff
@@ -23,6 +23,34 @@ class ItemHandler {
         return result == undefined ? true : result;
     }
 
+    public static useAllBattleItems() {
+        GameHelper.chunk(6, Object.keys(ItemList).filter(i => ItemList[i].constructor.name == 'BattleItem')).forEach((item,index) => {
+            item.forEach((x) => {
+                ItemHandler.useItem(x, EffectEngineRunner.amountToUse());
+            });
+        });
+    }
+
+    public static activateAllFlutes() {
+        GameHelper.chunk(6, Object.keys(ItemList).filter(i => ItemList[i].constructor.name == 'FluteItem')).forEach((item,index) => {
+            item.forEach((x) => {
+                if (this.hasItem(x) && !FluteEffectRunner.isActive(GameConstants.FluteItemType[x])()) {
+                    ItemHandler.useItem(x);
+                }
+            });
+        });
+    }
+
+    public static deactivateAllFlutes() {
+        GameHelper.chunk(6, Object.keys(ItemList).filter(i => ItemList[i].constructor.name == 'FluteItem')).forEach((item,index) => {
+            item.forEach((x) => {
+                if (FluteEffectRunner.isActive(GameConstants.FluteItemType[x])()) {
+                    ItemHandler.useItem(x);
+                }
+            });
+        });
+    }
```

Comment:
can be fuse on one function with boolean as parameter, and I would have personally put "item" of the foreach in the plural and replaced "x" by item in the singular.

but deactiveAllFlutes really deactive ? the useItem make me doubts :thinking:

Reply:
Yes it really deactivate, it confused me too at first. I will make the other corrections you mentioned.